tests/integ/test_async_inference.py

```diff
@@ -0,0 +1,97 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import pytest
+import os
+
+import sagemaker.amazon.pca
+from sagemaker.utils import unique_name_from_base
+from sagemaker.async_inference import AsyncInferenceConfig, AsyncInferenceResponse
+from sagemaker.predictor_async import AsyncPredictor
+from tests.integ import datasets, TRAINING_DEFAULT_TIMEOUT_MINUTES
+from tests.integ.timeout import timeout, timeout_and_delete_endpoint_by_name
+
+INPUT_LOCAL_PATH = "tests/data/async_inference_input/async-inference-pca-input.csv"
+
+
+@pytest.fixture
+def training_set():
+    return datasets.one_p_mnist()
+
+
+def test_async_walkthrough(sagemaker_session, cpu_instance_type, training_set):
```

Comment:
nit: one integ test is enough? do we need any other end user case scenerio we need to test here.

Reply:
I think this should be enough, cause in this one integ test we test all possible use cases to invoke predict method (wait/not_wait/input_data/s3_location).